origin-js/src/resources/marketplace.js

```diff
@@ -118,14 +118,14 @@ class Marketplace {
       const listingPrice = await this.contractService.moneyToUnits(listing.price)
       const listingCommision = await this.contractService.moneyToUnits(listing.commission)
       const currencies = await this.contractService.currencies()
-      const currency = currencies[listingCurrency]
+      const currency = listingCurrency &&currencies[listingCurrency]
```

Comment:
Yes, good point - I can revisit this later today

Reply:
@franckc I separated out the checks for `listing.price`, which only apply to `unit` listings, and I added a TODO because I realized that we should be checking fractional listings to make sure the currency of the offer is the same as the currency of the listing, but since we don't have a `price` object on fractional listings (since the price is defined in each slot), we'll need to add a way to track the currency of a fractional listing and then do that check. 

I also think it might make sense to add some similar checks to `makeOffer` at some point so it's doing this kind of validation before the offer is created...